File: Medium/0240-Search a 2D Matrix II/Solution.kt
class Solution {
    fun searchMatrix(matrix: Array<IntArray>, target: Int): Boolean {
        if (matrix.isEmpty() || matrix[0].isEmpty()) {
            return false
        }

        var row = matrix.size - 1
        var col = 0

        while (row >= 0 && col < matrix[0].size) {
            if (matrix[row][col] < target) {
                col += 1
            } else if (matrix[row][col] > target) {
                row -= 1
            } else {
                return true
            }
        }

        return false
    }
}

File: Medium/0240-Search a 2D Matrix II/Solution.rb
# @param {Integer[][]} matrix
# @param {Integer} target
# @return {Boolean}
def search_matrix(matrix, target)
    return false if matrix.empty? or matrix[0].empty?

    row, col = matrix.length - 1, 0

    while row >= 0 and col < matrix[0].length
        if matrix[row][col] < target
            col += 1
        elsif matrix[row][col] > target
            row -= 1
        else
            return true
        end
    end

    return false
end
